src/jdk.incubator.concurrent/share/classes/jdk/incubator/concurrent/ScopedValue.java

```diff
@@ -43,71 +42,109 @@
+ * is bound to some value may be bound to a new value for the bounded execution of some
+ * method. The unfolding execution of code executed by that method defines the nested
+ * dynamic scope. When the method completes (normally or with an exception), the value of
+ * the {@code ScopedValue} reverts to its previous value.
  *
+ * <p> In the above example, suppose that code executed by {@code doSomething()} binds
+ * {@code USERNAME} to a new value with:
  * {@snippet lang=java :
- *   // @link substring="newInstance" target="ScopedValue#newInstance" :
- *   private static final ScopedValue<Credentials> CREDENTIALS = ScopedValue.newInstance();
+ *     ScopedValue.where(USERNAME, "duchess", () -> doMore());
+ * }
+ * Code executed directly or indirectly by {@code doMore()} that invokes {@code
+ * USERNAME.get()} will read the value "{@code duchess}". When {@code doMore()} completes
+ * (normally or with an exception), the value of {@code USERNAME} reverts to
+ * "{@code duke}".
+ *
+ * <h2><a id="inheritence">Inheritance</a></h2>
+ *
+ * {@code ScopedValue} supports sharing data across threads. At this time, this sharing
+ * is limited to structured cases where child threads are started and terminate within
+ * the bounded period of execution by a parent thread. More specifically, when using a
+ * {@link StructuredTaskScope}, scoped value bindings are <em>captured</em> when creating
+ * a {@code StructuredTaskScope} and inherited by all threads started in that scope with
+ * the {@link StructuredTaskScope#fork(Callable) fork} method.
  *
- *   Credentials creds = ...
- *   ScopedValue.where(CREDENTIALS, creds).run(() -> {
- *       ...
- *       Connection connection = connectDatabase();
- *       ...
- *   });
+ * <p> In the following example, the {@code ScopedValue} {@code USERNAME} is bound to the
+ * value "{@code duke}" for the execution of a runnable operation. The code in the {@code
+ * run} method creates a {@code StructuredTaskScope} and forks three child threads. Code
+ * executed directly or indirectly by these threads running {@code childTask1()},
+ * {@code childTask2()}, and {@code childTask3()} will read the value "duke".
  *
- *   ...
+ * {@snippet lang=java :
+ *     private static final ScopedValue<String> USERNAME = ScopedValue.newInstance();
+
+ *     ScopedValue.where(USERNAME, "duke", () -> {
+ *         try (var scope = new StructuredTaskScope<String>()) {
  *
- *   Connection connectDatabase() {
- *       // @link substring="get" target="ScopedValue#get" :
- *       Credentials credentials = CREDENTIALS.get();
- *       ...
- *   }
+ *             scope.fork(() -> childTask1());
+ *             scope.fork(() -> childTask2());
+ *             scope.fork(() -> childTask3());
+ *
+ *             ...
+ *          }
+ *     });
  * }
  *
+ * <p> Unless otherwise specified, passing a {@code null} argument to a method in this
+ * class will cause a {@link NullPointerException} to be thrown.
+ *
```

Comment:
Should this last sentence be elsewhere? It's rather lost at the end of the "Inheritance" section.

Reply:
The insertion of the Rebinding and Inheritance headings causds it to get pushed down, making it look like its in the Inheritance section.